test/onnx/torch_export/test_torch_export_with_onnxruntime.py

```diff
@@ -0,0 +1,298 @@
+
+        test_inputs = (torch.randn(4, 4), torch.randn(7, 4))
+        test_kwargs = {"kw2": torch.ones(4, 4), "kw1": torch.zeros(9, 4)}
+        # This should work even if the kwarg order are flipped.
+        self._compare_onnx_and_torch_exported_program(
+            exported_program, onnx_program, test_inputs, test_kwargs
+        )
+
+    def test_exported_program_as_input_lifting_buffers_mutation(self):
+        for persistent in (True, False):
+
+            class CustomModule(torch.nn.Module):
+                def __init__(self):
+                    super().__init__()
+                    self.register_buffer(
+                        "my_buffer", torch.tensor(4.0), persistent=persistent
+                    )
+
+                def forward(self, x, b):
+                    output = x + b
+                    (
+                        self.my_buffer.add_(1.0) + 3.0
+                    )  # Mutate buffer through in-place addition
+                    return output
+
+            input_x = torch.rand((3, 3), dtype=torch.float32)
+            input_b = torch.randn(3, 3)
+            model = CustomModule()
+
+            dim = torch.export.Dim("dim")
+            exported_program = torch.export.export(
+                model,
+                (
+                    input_x,
+                    input_b,
+                ),
+                dynamic_shapes=({0: dim}, {0: dim}),
+            )
+            onnx_program = torch.onnx.dynamo_export(exported_program, input_x, input_b)
+
+            # different dim inputs
+            additional_inputs_x = torch.rand((4, 3), dtype=torch.float32)
+            additional_inputs_b = torch.randn(4, 3)
+            self._compare_onnx_and_torch_exported_program(
+                exported_program,
+                onnx_program,
+                (
+                    additional_inputs_x,
+                    additional_inputs_b,
+                ),
+            )
+
+    def test_onnx_program_supports_non_arg_name_with_container_type(self):
+        def foo(a, b):
+            return a[0].sum() + a[1].sum() + b.sum()
+
+        inp_a = (torch.randn(4, 4), torch.randn(4, 4))
+        inp_b = torch.randn(4, 4)
+        inp = (inp_a, inp_b)
+
+        count = 0
+
+        def dynamify_inp(x):
+            # Mark the second input a[1] dynamic
+            nonlocal count
+            if count == 1:
+                dim = torch.export.Dim("dim", min=3)
+                count += 1
+                return {0: dim}
+            count += 1
+            return None
+
+        dynamic_shapes = torch_pytree.tree_map(dynamify_inp, inp)
+        exported_program = torch.export.export(foo, inp, dynamic_shapes=dynamic_shapes)
+        onnx_program = torch.onnx.dynamo_export(exported_program, inp_a, inp_b)
+
+        # NOTE: Careful with the input format. The input format should be
+        # consistent with how the model is exported.
+        test_inputs = ((torch.randn(4, 4), torch.randn(6, 4)), torch.randn(4, 4))
```

Comment:
you should use `onnx_program.adapt_torch_inputs_tyo_onnx()` to adapt the "normal torch input" to the format expected by the onnx model

Reply:
Not sure about this. The input is adpated inside onnx_program __call__.